sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2522,19 +2522,6 @@ object SQLConf {
       .booleanConf
       .createWithDefault(false)
 
-  val LEGACY_PARQUET_REBASE_DATETIME_IN_WRITE =
-    buildConf("spark.sql.legacy.parquet.rebaseDateTimeInWrite.enabled")
```

Comment:
One thing I am concerned, though, I suspect it won't be compatible with other external projects if we don't rebase?  We have `spark.sql.parquet.writeLegacyFormat` in particular for the compatibility with Apache Hive and Apache Impala for example.

Reply:
People would assume the storage systems to use the ISO standard calendar (proleptic Gregorian calendar) by default. The Parquet spec doesn't define the calendar explicitly but it refers to java 8 time API, which kind of implictly requires proleptic Gregorian calendar.

Spark has this issue because we use java 7 time API before 3.0.

Hive switched to the proleptic Gregorian calendar in 3.1, and AFAIK caused some troubles in other systems, which kind of proves that other systems are using the proleptic Gregorian calendar already.

Impala has added a config `convert_legacy_hive_parquet_utc_timestamps` to work around it: https://issues.apache.org/jira/browse/IMPALA-3933

Presto also hit issues when reading Hive tables with timestamp: https://github.com/prestodb/presto/issues/12180

That why I think it's better to write datetime values without rebasing, as that's the correct data.

Reply:
Thanks for investigation. SGTM.